app/graphql/mutations/post/unlock_post.rb

```diff
@@ -0,0 +1,31 @@
+class Mutations::Post::UnlockPost < Mutations::Base
+  argument :input,
+    Types::Input::Post::Unlock,
+    required: true,
+    description: 'Unlock a Post.',
+    as: :post
+
+  field :post, Types::Post, null: true
+
+  def load_post(value)
+    post = ::Post.find(value.id)
+    post.assign_attributes(value.to_model)
+    post
+  end
+
+  def authorized?(post:)
+    super(post, :update_lock?)
+  end
+
+  def resolve(post:)
+    post.save
+
+    if post.errors.any?
+      Errors::RailsModel.graphql_error(post)
+    else
+      {
+        post: post
+      }
+    end
```

Comment:
FWIW I added an `Errors::ActiveRecord::RecordInvalid` in #880 that should do this for us in future with `post.save!`

We can change it once both are merged